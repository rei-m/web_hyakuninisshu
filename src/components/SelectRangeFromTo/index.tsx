import * as React from 'react';
import { TextAlignProperty } from 'csstype';
import withStyles from '@material-ui/core/styles/withStyles';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';
import MenuItem from '@material-ui/core/MenuItem';

import styled from '@src/styles/styled-components';
import { RangeFromConditions, RangeToConditions } from '@src/enums';
import { toKarutaNoString } from '@src/utils';

export interface Props {
  from: string;
  to: string;
  fromTouched?: boolean;
  toTouched?: boolean;
  error?: string;
  style?: React.CSSProperties;
  handleChange: (e: React.SyntheticEvent<HTMLSelectElement>) => void;
}

const styles = {
  formControl: {
    display: 'flex',
  },
  select: {
    fontSize: '1.6rem',
    maxWidth: '150px',
    flexGrow: 1,
    textAlign: 'left' as TextAlignProperty,
  },
};

const stylesProvider = () => styles;

type RenderProps = Props & {
  classes: {
    formControl: string;
    select: string;
  };
};

const FormRow = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`;

const Label = styled.label`
  text-align: left;
  font-size: 1.1rem;
  color: #182026;
`;

const Separate = styled.span`
  padding: ${({ theme }) => theme.spacing2x};
`;

const Error = styled.div`
  font-size: 1.2rem;
  color: #f00;
  margin: ${({ theme }) => `${theme.spacing1x} 0`};
`;

const SelectRangeFromTo = withStyles(stylesProvider)(
  ({ from, to, error, handleChange, classes, style }: RenderProps) => (
    <FormControl className={classes.formControl} style={style}>
      <Label>出題範囲</Label>
      <FormRow>
        <Select
          value={from}
          onChange={handleChange}
          inputProps={{
            name: 'rangeFrom',
            id: `id-range-from`,
          }}
          className={classes.select}
        >
          {RangeFromConditions.values.map(value => (
            <MenuItem value={value} className={classes.select} key={`range_from_${value}`}>
              {toKarutaNoString(value)}
            </MenuItem>
          ))}
        </Select>
        <Separate>〜</Separate>
        <Select
          value={to}
          onChange={handleChange}
          inputProps={{
            name: 'rangeTo',
            id: `id-range-to`,
          }}
          className={classes.select}
        >
          {RangeToConditions.values.map(value => (
            <MenuItem value={value} className={classes.select} key={`range_to_${value}`}>
              {toKarutaNoString(value)}
            </MenuItem>
          ))}
        </Select>
      </FormRow>
      {error && <Error>{error}</Error>}
    </FormControl>
  )
);

export default SelectRangeFromTo;
