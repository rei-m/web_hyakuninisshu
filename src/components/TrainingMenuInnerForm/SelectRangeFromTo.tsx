import * as React from 'react';
import styled from 'styled-components';
import { Field } from 'formik';
import { RANGE_FROM, RANGE_TO } from '../../constants/Training';

const Label = styled.label`
  text-align: left;
`;

const SelectRow = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;

const SelectRange = styled.div`
  width: 150px;
`;

const Separate = styled.span`
  padding: 16px;
`;

const Error = styled.div`
  font-size: 1.2rem;
  color: #f00;
  margin: 8px 16px;
`;

export interface SelectRangeFromToProps {
  readonly from: string;
  readonly to: string;
  readonly fromTouched?: boolean;
  readonly toTouched?: boolean;
  readonly error?: string;
  readonly handleChange: (e: React.SyntheticEvent<HTMLSelectElement>) => void;
}

const SelectRangeFromTo = ({
  from,
  to,
  fromTouched,
  toTouched,
  error,
  handleChange
}: SelectRangeFromToProps) => (
  <Label className="pt-label">
    出題範囲
    <SelectRow>
      <SelectRange className="pt-select pt-large">
        <Field
          component="select"
          name="rangeFrom"
          value={from}
          onChange={handleChange}
        >
          {RANGE_FROM.map((item, i) => (
            <option value={item.value} key={`range_from_${i}`}>
              {item.name}
            </option>
          ))}
        </Field>
      </SelectRange>
      <Separate>〜</Separate>
      <SelectRange className="pt-select pt-large">
        <Field
          component="select"
          name="rangeTo"
          value={to}
          onChange={handleChange}
        >
          {RANGE_TO.map((item, i) => (
            <option value={item.value} key={`range_to_${i}`}>
              {item.name}
            </option>
          ))}
        </Field>
      </SelectRange>
    </SelectRow>
    {(fromTouched || toTouched) && error && <Error>{error}</Error>}
  </Label>
);

export default SelectRangeFromTo;
