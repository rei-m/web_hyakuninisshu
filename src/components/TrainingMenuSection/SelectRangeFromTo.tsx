import * as React from 'react';
import styled from 'styled-components';
import { RANGE_FROM, RANGE_TO } from '../../constants/Training';

const Label = styled.label`
  text-align: left;
`;

const SelectRow = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;

const SelectRange = styled.div`
  width: 150px;
`;

const Separate = styled.span`
  padding: 16px;
`;

export interface SelectRangeFromToProps {
  fromValue: number;
  toValue: number;
}

const onChangeFrom = (e: React.SyntheticEvent<HTMLSelectElement>) => {
  console.dir(e);
};

const onChangeTo = (e: React.SyntheticEvent<HTMLSelectElement>) => {
  console.dir(e);
};

const SelectRangeFromTo = (props: SelectRangeFromToProps) => (
  <Label className="pt-label">
    出題範囲
    <SelectRow>
      <SelectRange className="pt-select pt-large">
        <select value={props.fromValue} onChange={onChangeFrom}>
          {RANGE_FROM.map((item, i) => (
            <option value={item.value} key={`range_from_${i}`}>
              {item.name}
            </option>
          ))}
        </select>
      </SelectRange>
      <Separate>〜</Separate>
      <SelectRange className="pt-select pt-large">
        <select value={props.toValue} onChange={onChangeTo}>
          {RANGE_TO.map((item, i) => (
            <option value={item.value} key={`range_to_${i}`}>
              {item.name}
            </option>
          ))}
        </select>
      </SelectRange>
    </SelectRow>
  </Label>
);

export default SelectRangeFromTo;
