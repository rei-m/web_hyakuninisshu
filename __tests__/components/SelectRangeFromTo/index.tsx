import * as ReactTestRenderer from 'react-test-renderer';
import * as React from 'react';
import SelectRangeFromTo, { Props } from '@src/components/molecules/SelectRangeFromTo';

describe('<SelectRangeFromTo />', () => {
  it('should render component', () => {
    const props: Props = {
      from: 'from',
      to: 'to',
      handleChange: jest.fn(),
    };
    const renderer = ReactTestRenderer.create(<SelectRangeFromTo {...props} />);
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
