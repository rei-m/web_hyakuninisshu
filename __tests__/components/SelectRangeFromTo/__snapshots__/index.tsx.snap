// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectRangeFromTo /> should render component 1`] = `
<div
  className="MuiFormControl-root-3 Component-formControl-1"
>
  <label
    className="sc-bwzfXH jjTplH"
  >
    出題範囲
  </label>
  <div
    className="sc-bdVaJa grbsvW"
  >
    <div
      className="MuiInputBase-root-27 MuiInput-root-14 MuiInput-underline-18 MuiInputBase-formControl-28 MuiInput-formControl-15 Component-select-2"
      onClick={[Function]}
    >
      <div
        className="MuiSelect-root-7"
      >
        <div
          aria-haspopup="true"
          aria-pressed="false"
          className="MuiSelect-select-8 MuiSelect-selectMenu-11 MuiInputBase-input-37 MuiInput-input-22"
          id="select-rangeFrom"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </div>
        <input
          id="id-range-from"
          name="rangeFrom"
          type="hidden"
          value="from"
        />
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-44 MuiSelect-icon-13"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
    <span
      className="sc-htpNat dDpdjb"
    >
      〜
    </span>
    <div
      className="MuiInputBase-root-27 MuiInput-root-14 MuiInput-underline-18 MuiInputBase-formControl-28 MuiInput-formControl-15 Component-select-2"
      onClick={[Function]}
    >
      <div
        className="MuiSelect-root-7"
      >
        <div
          aria-haspopup="true"
          aria-pressed="false"
          className="MuiSelect-select-8 MuiSelect-selectMenu-11 MuiInputBase-input-37 MuiInput-input-22"
          id="select-rangeTo"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </div>
        <input
          id="id-range-to"
          name="rangeTo"
          type="hidden"
          value="to"
        />
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-44 MuiSelect-icon-13"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
